### PR TITLE
[ac][pattern matcher] Do not percolate tags beyond the inputs of matched portion

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #118034

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: test/dynamo/test_activation_checkpointing.py [hunk 1]
```diff
@@ -32,12 +40,12 @@ def match_rng_op(node, op):
                 return node.args[1] == op
         return False
 
-    assert ((freq or freq_ge) and op) or ((freqs or freqs_ge) and ops)
-    if op:
+    # assert ((freq or freq_ge) and op) or ((freqs or freqs_ge) and ops)
```
Comment (yf225) [inline test/dynamo/test_activation_checkpointing.py:43]: curious: why do we need to remove it?

**Review by yf225 — APPROVED** [review]

File: test/dynamo/test_activation_checkpointing.py (hunk 1 above)
Comment (anijain2305) [reply test/dynamo/test_activation_checkpointing.py:43]: I wanted to have freq=0 to check that an op is absent in the graph. assert was failing there.

**Pull request closed without merging**